include/deal.II/matrix_free/tools.h

```diff
@@ -2221,11 +2221,15 @@ namespace MatrixFreeTools
 
                   for (unsigned int v = 0; v < n_filled_lanes; ++v)
                     for (unsigned int bi = 0; bi < n_blocks; ++bi)
-                      constraints.distribute_local_to_global(
-                        matrices[bi][bj][v],
-                        dof_indices_mf[bi][v],
-                        dof_indices_mf[bj][v],
-                        matrix);
+                      if (bi == bj)
+                        constraints.distribute_local_to_global(
+                          matrices[bi][bi][v], dof_indices_mf[bi][v], matrix);
+                      else
+                        constraints.distribute_local_to_global(
+                          matrices[bi][bj][v],
+                          dof_indices_mf[bi][v],
+                          dof_indices_mf[bj][v],
+                          matrix);
```

Comment:
I agree with this change, but we should document why we choose different argument lists. We do have a quite extensive documentation in `AffineConstraints` as I indicated in #17140, but it is easy to miss it when looking at the use of `distribute_local_to_global` function, as both the code writer and rewiewers of #17140 missed it.

Reply:
@kronbichler I have added a comment!